src/components/VoiceSearch/__tests__/VoiceSearch-test.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import VoiceSearch, { VoiceSearchProps } from '../VoiceSearch';
+
+const defaultProps: VoiceSearchProps = {
+  cssClasses: {
+    root: 'root',
+    button: 'button',
+    status: 'status',
+  },
+  isBrowserSupported: true,
+  isListening: false,
+  toggleListening: () => {},
+  voiceListeningState: {
+    status: 'initial',
+  },
+  templates: {
+    buttonText: 'button',
+    status: 'status',
+  },
+};
+
+describe('VoiceSearch', () => {
+  describe('button', () => {
+    it('call toggleListening when button is clicked', () => {
+      const props = {
+        ...defaultProps,
+        toggleListening: jest.fn(),
+      };
+      const wrapper = mount(<VoiceSearch {...props} />);
+      wrapper.find('button').simulate('click');
+      expect(props.toggleListening).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Rendering', () => {
+    it('with default props', () => {
+      expect(mount(<VoiceSearch {...defaultProps} />)).toMatchSnapshot();
+    });
+
+    it('button disabled in unsupported browser', () => {
+      const props = {
+        ...defaultProps,
+        isBrowserSupported: false,
+      };
+      const wrapper = mount(<VoiceSearch {...props} />);
+      expect(wrapper.find('button').props().disabled).toBe(true);
+    });
+
+    it('with custom template for buttonText (1)', () => {
+      const props = {
+        ...defaultProps,
+        isListening: true,
+        templates: {
+          buttonText: ({ isListening }) => (isListening ? 'Stop' : 'Start'),
+          status: ``,
+        },
+      };
+      expect(mount(<VoiceSearch {...props} />)).toMatchSnapshot();
```

Comment:
We can get rid of the overall snapshot and test the actual content of the template. It makes the intent of the test clearer:

```js
const wrapper = mount(<VoiceSearch {...props} />);
expect(wrapper.find('.button').text()).toBe('Stop');
```